Dev make cleanup

TerryE opened the pull request:
Fixes #2839

- [x] This PR is for the `dev` branch rather than for `master`.
- [x] This PR is compliant with the [other contributing guidelines](https://github.com/nodemcu/nodemcu-firmware/blob/dev/CONTRIBUTING.md) as well (if not, please describe why).
- [x] I have thoroughly tested my contribution.

This change doesn't affect the compiled binary, just the make process itself.  The best way to see this is with a `make V=1` before and after.

nwf commented:
These changes all seem sensible.  While you're at it, can the `** HEALTH WARNING **` section of the main `Makefile` get tidied, too?  (Perhaps that's a bigger thing and should wait for a later PR.)

TerryE commented:
OK I've now found why Travis failed.  The build process specifies EXTRA_CCFLAGS as a commandline parameter and doing this overrides all EXTRA_CCFLAGS setting inside the make so I need to make some adjustments to accommodate this.

TerryE commented:
It might be good to get #2841 and this in before we do the master drop, but I think that they both touch some common files, so whoever goes second will need to do some fixups or rebaseline.  We should let #2841 go first and soon, I think.

jmattsson commented on app/Makefile:
```diff
@@ -54,35 +52,36 @@ endif # } PDIR
 	uzlib/libuzlib.a 			\
 	$(OPT_SEL_COMPONENTS)
-
+# Ensure that the Espresif SDK is searched before the tool-chain's SDK (if any)
```
I don't think this comment is accurate here. Copy paste slip?

TerryE replied:
I will check and remove

jmattsson commented on app/user/Makefile:
```diff
@@ -38,13 +38,7 @@ DEFINES += -DESP_INIT_DATA_DEFAULT="\"$(SDK_DIR)/bin/esp_init_data_default_v05.b
-INCLUDES += -I ../platform
-INCLUDES += -I ../lua
-INCLUDES += -I ../spiffs
+INCLUDES += -I ../../include/ets -I ../spiffs
```
I don't think we have an `include/ets` either, we should be able to evict that too.

TerryE replied:
Will do

jmattsson commented on sdk-overrides/include/c_types.h:
```diff
@@ -0,0 +1,2 @@
+#define ICACHE_FLASH
+#include_next "c_types.h"
```
Objection, your honour! :)
When I hit that other shiny green button just before, our `c_types.h` got decimated and should stay that way.

TerryE replied:
Yup, agreed. All part of the rebaseline TBD today. The reason for the wrapper is that ICACHE_FLASH was only used in this include.

TerryE commented:
I will do the rebaseline my late PM. I've got other commitments before then.

TerryE commented:
I am note sure of the procedural niceties here. @nwf has reviewed this but not used the github review function do to this.  @jmattsson has also reviewed this and I've addressed these comments but not I am not sure how to mark this off as completed.  Anyway, if this Travis make completes then this is a verification of the make options as reviewed by two other committers, so I will then merge and squash this.
